### DISCUSSION TITLE
Docsearch integration issue

### DISCUSSION THREAD

**Discussion by TheDrone7** [vuepress/core#911] [Q&A] [answered]
Hello, I was using the docsearch plugin https://v2.vuepress.vuejs.org/reference/plugin/docsearch.html#usage

I registered my site and they shared with me the App ID and other details, and I added the following to my `config.ts`, the source can be viewed [here](https://github.com/TheDrone7/shieldbow/blob/docsearch/docs/.vuepress/config.ts) (the `docsearch` branch)

```ts
plugins: [
    docsearchPlugin({
      appId: '<id>',
      apiKey: '<api-key>',
      indexName: '<index-name>'
    }),
  ],
```

(replacing all the placeholders) it neither seems to be registering the docsearch component nor does it add it to the navbar (I am using the default theme). There are no errors either. Is there an implementation issue?

**Comment by meteorlxy** [comment, marked as answer]
You should provide a reproduce repo, or we could not help as the information is so limited.

**Reply by TheDrone7** [reply]
Updated the question to include the source repo

**Reply by meteorlxy** [reply]
Please upgrade your vuepress version first. You are using beta.43 core with beta.45 plugin

**Reply by Mister-Hope** [reply]
I think we'd better use exact version untill we make sure no more breaking changes and move to rc. @meteorlxy

**Reply by Mister-Hope** [reply]
Oh, just found out the original version is "workspace:*", I am not sure, does pnpm support configuring this to make the version be exact when publishing?

**Reply by meteorlxy** [reply]
We are using exact version dude 😅

**Reply by Mister-Hope** [reply]
Emm, shall we declare peerDeps and mark them optional? just like sass-loader.

The problem is that when users are using a plugin version different from core, he does not get any error hints.

**Reply by Mister-Hope** [reply]
Maybe we can do something to make a improvement here. A lot of my users will accidently install mutiple `@vuepress/client` and `@vuepress/core` in their repo.

Vue used to have these kind of problems, so vue is actually bundling some vue packages, so that `@vue/xxx` can be accessed and is recommanded to be accessed through `vue/xxx`, so I am actually thinking about things we can do.

**Reply by TheDrone7** [reply]
Updating the versions seems to have added it to the site but when I tried searching for stuff, no results popped up. I checked the search logs on algolia dashboard and even though, its getting requests, there seems to be no query in the requests.

**Reply by TheDrone7** [reply]
NVM that was a crawler config issue.
